tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5747,6 +5753,132 @@ SpirvEmitter::processTextureSampleCmp(const CXXMemberCallExpr *expr) {
       expr->getCallee()->getLocStart(), expr->getSourceRange());
 }
 
+SpirvInstruction *
+SpirvEmitter::processTextureSampleCmpBias(const CXXMemberCallExpr *expr) {
+  // .SampleCmpBias() Signature:
+  //
+  // For Texture1D, Texture1DArray, Texture2D, Texture2DArray:
+  // float Object.SampleCmpBias(
+  //   SamplerComparisonState S,
+  //   float Location,
+  //   float Bias,
+  //   float CompareValue
+  //   [, int Offset]
+  //   [, float Clamp]
+  //   [, out uint Status]
+  // );
+  //
+  // For TextureCube and TextureCubeArray:
+  // float Object.SampleCmpBias(
+  //   SamplerComparisonState S,
+  //   float Location,
+  //   float Bias,
+  //   float CompareValue
+  //   [, float Clamp]
+  //   [, out uint Status]
+  // );
+
+  const auto numArgs = expr->getNumArgs();
+  const bool hasStatusArg =
+      expr->getArg(numArgs - 1)->getType()->isUnsignedIntegerType();
+  auto *status = hasStatusArg ? doExpr(expr->getArg(numArgs - 1)) : nullptr;
+
+  SpirvInstruction *clamp = nullptr;
+  if (numArgs > 4 && expr->getArg(4)->getType()->isFloatingType())
+    clamp = doExpr(expr->getArg(4));
+  else if (numArgs > 5 && expr->getArg(5)->getType()->isFloatingType())
+    clamp = doExpr(expr->getArg(5));
```

Comment:
Determining which argument is which by looking at the types is very hard to read. Could you do something with the type of `this`? You can get it from `expr->getImplicitObjectArgument()->getType()`.

Reply:
In this case the type of `this` doesn't give us much useful information. The only difference is whether or not `offset` may be present. I've refactored the code to try to match the optional arguments in order – I think this approach is more readable.

The new helper method could potentially be used in the other texture sampling methods as well.